dev/run-tests

```diff
@@ -17,239 +17,394 @@
+    if os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE"):
+        os.environ[sbt_maven_profile_args_env] = \
+            " ".join(sbt_maven_profile_arg_dict.get(ajbp, []) 
+                     + sbt_maven_profile_args_base)
+    else:
+        os.environ[sbt_maven_profile_args_env] = \
+            " ".join(sbt_maven_profile_arg_dict.get("hadoop2.3", [])
+                     + sbt_maven_profile_args_base)
+
+def is_exe(path):
+    """Check if a given path is an executable file
+    - from: http://stackoverflow.com/a/377028"""
+
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+def which(program):
+    """Find and return the given program by its absolute path or 'None'
+    - from: http://stackoverflow.com/a/377028"""
+
+    fpath, fname = os.path.split(program)
+
+    if fpath:
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ.get("PATH").split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+    return None
+
+def determine_java_executable():
+    """Will return the *best* path possible for a 'java' executable or `None`"""
+
+    java_home = os.environ.get("JAVA_HOME")
+
+    # check if there is an executable at $JAVA_HOME/bin/java
+    java_exe = which(os.path.join(java_home, "bin/java"))
+    # if the java_exe wasn't set, check for a `java` version on the $PATH
+    return java_exe if java_exe else which("java")
+
+def determine_java_version(java_exe):
+    """Given a valid java executable will return its version in tuple format as:
+    [<major-version>, <minor-version>, <patch-version>, <update-version>]"""
+
+    raw_output = sp.check_output([java_exe, "-version"], stderr=sp.STDOUT)
+    raw_version_str = raw_output.split('\n')[0] # eg 'java version "1.8.0_25"'
+    version_str = raw_version_str.split()[-1].strip('"') # eg '1.8.0_25'
+    version, update = version_str.split('_') # eg ['1.8.0', '25']
+
+    # map over the values and convert them to integers
+    return map(lambda x: int(x), version.split('.') + [update])
+
+def multi_starts_with(orig_str, *prefixes):
+    """Takes a string and an abritrary number of prefixes then checks the
+    original string for any of the possible prefixes passed in"""
+
+    for s in prefixes:
+        if orig_str.startswith(s):
+            return True
+    return False
+
+# This function current acts to determine if SQL tests need to be run in
+# addition to the core test suite *or* if _only_ SQL tests need to be run
+# as the git logs show that to be the only thing touched. In the future
+# this function will act more generically to help further segregate the
+# test suite runner (hence the function name).
+# @return a set of unique test names
+def determine_test_suite():
+    test_suite = list()
+
+    if amplab_jenkins:
+        sp.Popen(['git', 'fetch', 'origin', 'master:master']).wait()
```

Comment:
Shouldn't this perhaps instead be `subprocess.check_call(...)`? We want to raise an exception if this fails, no? Right now it will just keep on truckin' even if the call to git fails.